modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/events/LogoutPreAction.java

```diff
@@ -44,18 +47,33 @@ public void run(
 			ProductNavigationControlMenuEntryConstants.SESSION_CLICKS_KEY,
 			"closed");
 
-		_segmentsExperimentSegmentsExperienceRequestProcessor.
-			cleanCookieLogoutAction(httpServletRequest, httpServletResponse);
+		Cookie[] cookies = httpServletRequest.getCookies();
+
+		if (ArrayUtil.isEmpty(cookies)) {
+			return;
+		}
+
+		for (Cookie cookie : cookies) {
+			if (StringUtil.startsWith(
+					cookie.getName(), _AB_TEST_VARIANT_ID_COOKIE_PREFIX)) {
+
+				_cookiesManager.deleteCookies(
+					_cookiesManager.getDomain(httpServletRequest),
+					httpServletRequest, httpServletResponse, cookie.getName());
+			}
+		}
 	}
 
+	private static final String _AB_TEST_VARIANT_ID_COOKIE_PREFIX =
```

Comment:
@regiCesar21 I missed this one :(

Please extract it to a public constant in a common place so you don't need to declare it in both classes.